src/grt/src/Grid.cpp

```diff
@@ -197,4 +197,12 @@ odb::Rect Grid::getGridArea() const
       lower_left_x_, lower_left_y_, upper_right_x_, upper_right_y_);
 }
 
+void Grid::addObstruction(int layer,
+                          const odb::Rect& obstruction,
+                          int min_layer,
+                          int max_layer)
+{
+  obstructions_[layer].push_back(obstruction);
+}
```

Comment:
I think you forgot the check

Reply:
My bad. I've updated it now to do the check outside Grid::addObstruction. In this way, I can avoid computing the geometries of obstructions that will never be considered. I will push it soon.